receiver/splunksearchapireceiver/receiver.go

```diff
@@ -201,3 +241,26 @@ func (ssapir *splunksearchapireceiver) getSplunkSearchResults(sid string, offset
 	}
 	return resp, nil
 }
+
+func (ssapir *splunksearchapireceiver) checkpoint(ctx context.Context) error {
+	marshalBytes, err := json.Marshal(ssapir.checkpointRecord)
+	if err != nil {
+		return fmt.Errorf("failed to write checkpoint: %w", err)
+	}
+	return ssapir.storageClient.Set(ctx, eventStorageKey, marshalBytes)
+}
+
+func (ssapir *splunksearchapireceiver) loadCheckpoint(ctx context.Context) {
+	marshalBytes, err := ssapir.storageClient.Get(ctx, eventStorageKey)
+	if err != nil {
+		ssapir.logger.Error("failed to read checkpoint", zap.Error(err))
+		return
+	}
+	if marshalBytes == nil {
+		ssapir.logger.Info("no checkpoint found")
+		return
+	}
+	if err = json.Unmarshal(marshalBytes, ssapir.checkpointRecord); err != nil {
```

Comment:
shouldn't we bomb out if we fail to unmarshal a checkpoint?

Reply:
We shouldn't Fatal because that would stop the collector process entirely. I'd prefer seeing the error propogated back up. 

We do this so that other components can have clean Shutdown calls.